src/components/CardNumberInput/index.tsx

```diff
@@ -1,58 +1,39 @@
-import { useState } from 'react';
-
+import styled from 'styled-components';
 import Label from '../common/Label';
 import Input from '../common/Input';
-import { CARD_META_INFO, INPUT_RULES } from '../../constants/card-app';
 
-import styled from 'styled-components';
-import { VALIDATION_MESSAGES } from '../../constants/card-app';
+import { CARD_META_INFO, INPUT_RULES, VALIDATION_MESSAGES } from '../../constants/card-app';
+import { errorCaption } from '../../utils/errorCaption';
 
 interface CardNumberInputProps {
   cardNumbers: string[];
-  errorCaption: (text: string) => JSX.Element;
-  onCardNumberChange: (index: number, value: string) => void;
+  cardNumberErrors: boolean[];
+  onCardNumberChange: (e: React.ChangeEvent<HTMLInputElement>, index: number) => void;
 }
 
-const CardNumberInput = ({ cardNumbers, errorCaption, onCardNumberChange: handleCardNumberChange }: CardNumberInputProps) => {
-  const [inputErrors, setInputErrors] = useState<boolean[]>(Array.from<boolean>({ length: cardNumbers.length }).fill(false));
-
-  const handleInputChange = (index: number, value: string) => {
-    const isNumericInput = /^(\d*)$/.test(value);
-    const isValidateCardNumber = value.length === 4;
-
-    setInputErrors((prevErrors) => {
-      const newPrevErrors = [...prevErrors];
-      newPrevErrors[index] = !isNumericInput || !isValidateCardNumber;
-
-      return newPrevErrors;
-    });
-
-    if (!isNumericInput) return;
-
-    handleCardNumberChange(index, value);
-  };
-
-  const hasErrorInput = inputErrors.some((error) => error);
+const CardNumberInput = ({ cardNumbers, cardNumberErrors, onCardNumberChange }: CardNumberInputProps) => {
+  const hasError = cardNumberErrors.some((error) => error);
```

Comment:
(minor) 기존 코드도 딱히 문제 없지만, 아래와 같이 직관성을 좀 더 높여볼 수도 있을 것 같습니다!

```suggestion
  const hasError = cardNumberErrors.some(Boolean);
```

Reply:
Boolean 함수를 전달할 수도 있군요..! 처음 알았는데 훨씬 깔끔한 것 같습니다😮

`some`이나 `every`를 사용할 때 유용할 듯합니다 :)